src/Repository/IRepository.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Repository
+{
+    public interface IRepository<T> where T: IIdentifiable
```

Comment:
The main reason is that IRepository expresses exactly the contract my application needs, and methods can be added/removed as needed.  Using a framework interface like IDictionary removes this flexibility.

Another reason is that IRepository is an ordered list, while IDictionary is unordered.